localstack/dev/run/configurators.py

```diff
@@ -0,0 +1,407 @@
+
+    def __init__(
+        self,
+        *,
+        host_paths: HostPaths = None,
+        container_paths: ContainerPaths = None,
+        venv_path: Path = None,
+        pro: bool = False,
+    ):
+        self.host_paths = host_paths
+        self.pro = pro
+        # FIXME find a better way to determine the default venv
+        self.venv_path = venv_path or Path(os.path.join(os.getcwd(), ".venv"))
+        self.container_paths = container_paths or (
+            ProContainerPaths() if pro else CommunityContainerPaths()
+        )
+
+    def __call__(self, cfg: ContainerConfiguration):
+        # locate all relevant dependency directories
+        pattern = self.dependency_glob
+        files = _list_files_in_container_image(DOCKER_CLIENT, cfg.image_name)
+        paths = [PurePosixPath(f) for f in files]
+        # builds an index of "jinja2: /opt/code/.../site-packages/jinja2"
+        container_path_index = {p.name: p for p in paths if p.match(pattern)}
+
+        # find dependencies from the host
+        for dep_path in self.venv_path.glob("lib/python3.*/site-packages/*"):
+            # filter out everything that heuristically cannot be a source directory
+            if not dep_path.is_dir():
+                continue
+            if dep_path.name.endswith(".dist-info"):
+                continue
+            if dep_path.name == "__pycache__":
+                continue
+
+            if dep_path.name in container_path_index:
+                # find the target path in the index if it exists
+                target_path = str(container_path_index[dep_path.name])
+            else:
+                # if the given dependency is not in
+                target_path = self.container_paths.dependency_source(dep_path.name)
+
+            if self._has_mount(cfg.volumes, target_path):
+                continue
+
+            cfg.volumes.append(VolumeBind(str(dep_path), target_path))
+
+    def _has_mount(self, volumes: VolumeMappings, target_path: str) -> bool:
+        return True if volumes.find_target_mapping(target_path) else False
+
+
+def _list_files_in_container_image(container_client: ContainerClient, image_name: str) -> list[str]:
+    """
+    Uses ``docker export | tar -t`` to list all files in a given docker image. It caches the result based on
+    the image ID into a gziped file into ``~/.cache/localstack-dev-cli`` to (significantly) speed up
+    subsequent calls.
+
+    :param container_client: the container client to use
+    :param image_name: the container image to analyze
+    :return: a list of file paths
+    """
+    if not image_name:
+        raise ValueError("missing image name")
+
+    image_id = container_client.inspect_image(image_name)["Id"]
+
+    cache_dir = get_user_cache_dir() / "localstack-dev-cli"
+    cache_dir.mkdir(exist_ok=True, parents=True)
+    cache_file = cache_dir / f"{image_id}.files.txt.gz"
+
+    if not cache_file.exists():
```

Comment:
When would this cache file be updated? Is this a responsibility of the user, and should this be documented somewhere?

Reply:
i don't think it needs updating by the user. each image hash (unique) has its own cache file.

Reply:
You are absolutely right, completely read that wrong :)